src/ring_theory/integral_closure.lean

```diff
@@ -114,23 +114,39 @@ variables {R A B S : Type*}
 variables [comm_ring R] [comm_ring A] [comm_ring B] [comm_ring S]
 variables [algebra R A] [algebra R B] (f : R →+* S)
 
-theorem is_integral_alg_hom (f : A →ₐ[R] B) {x : A} (hx : is_integral R x) : is_integral R (f x) :=
-let ⟨p, hp, hpx⟩ :=
-hx in ⟨p, hp, by rw [← aeval_def, aeval_alg_hom_apply, aeval_def, hpx, f.map_zero]⟩
+lemma is_integral_alg_hom_class {C F : Type*} [comm_ring C] [algebra A B] [algebra R C]
```

Comment:
Could you similarly rename this to `is_integral_alg_hom`, thereby generalizing the existing lemma?

Reply:
I've forgotten this lemma, thanks!